src/EditorFeatures/VisualBasicTest/Completion/CompletionProviders/ObjectCreationCompletionProviderTests.vb

```diff
@@ -99,5 +100,132 @@ End Namespace
 
             Await VerifyItemExistsAsync(markup, "N.S")
         End Function
+
+        <Fact, Trait(Traits.Feature, Traits.Features.Completion)>
+        <WorkItem(4585, "https://github.com/dotnet/roslyn/issues/4585")>
+        Public Async Function TestGenericArguments_PredefinedTypes() As Task
+            Dim markup = <Text><![CDATA[
+Imports System
+
+Class A(Of T)
+End Class
+
+Class B
+  Private _a As A(Of Int32) = New $$
+End Class
+]]></Text>.Value
+
+            Await VerifyItemExistsAsync(markup, "A(Of Integer)",
+                options:=[Option](CodeStyleOptions.PreferIntrinsicPredefinedTypeKeywordInDeclaration, LanguageNames.VisualBasic, True))
+        End Function
+
+        <Fact, Trait(Traits.Feature, Traits.Features.Completion)>
+        <WorkItem(4585, "https://github.com/dotnet/roslyn/issues/4585")>
+        Public Async Function TestGenericArguments_FrameworkTypes() As Task
+            Dim markup = <Text><![CDATA[
+Imports System
+
+Class A(Of T)
+End Class
+
+Class B
+  Private _a As A(Of Integer) = New $$
+End Class
+]]></Text>.Value
+
+            Await VerifyItemExistsAsync(markup, "A(Of Int32)",
+                options:=[Option](CodeStyleOptions.PreferIntrinsicPredefinedTypeKeywordInDeclaration, LanguageNames.VisualBasic, False))
+        End Function
+
+        <Fact(Skip:="https://github.com/dotnet/roslyn/issues/13610"), Trait(Traits.Feature, Traits.Features.Completion)>
+        Public Async Function EscapeAwaitIdentifierInAsyncMethod() As Task
+            Dim text =
+<code><![CDATA[
+Class Await
+End Class
+
+Class C
+    Async Sub Foo()
+        Dim a As [Await] = New $$
+    End Sub
+End Class
+]]></code>.Value
+
+            Dim expected =
+<code><![CDATA[
+Class Await
+End Class
+
+Class C
+    Async Sub Foo()
+        Dim a As [Await] = New [Await]
+    End Sub
+End Class
+]]></code>.Value
+
+            Await VerifyProviderCommitAsync(text, "Await", expected, Nothing, "")
```

Comment:
same comment about commit tests.  These should never be done inside provider tests.  Always in the CommandHandler tests.